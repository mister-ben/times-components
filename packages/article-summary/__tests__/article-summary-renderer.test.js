/* eslint-env jest */

import "react-native";
import renderer from "react-test-renderer";
import ArticleSummaryRenderer from "../article-summary-renderer";

it("renders a sentence ending with a full stop and a space with no new line", () => {
  const output = renderer
    .create(ArticleSummaryRenderer.sentence("some-key", {}, "Content"))
    .toJSON();
  expect(output).toMatchSnapshot();
});

it("renders a teaser with an ellipsis at the end", () => {
  const output = renderer
    .create(ArticleSummaryRenderer.teaser("some-key", {}, "Content"))
    .toJSON();

  expect(output).toMatchSnapshot();
});
