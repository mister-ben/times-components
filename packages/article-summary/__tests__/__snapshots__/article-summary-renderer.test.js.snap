// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a paragraph ending with a space with no new line 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Content
   
</Text>
`;

exports[`renders a sentence ending with a full stop and a space with no new line 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Content
  . 
</Text>
`;

exports[`renders a teaser with an ellipsis at the end 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  Content
  ...
</Text>
`;
